src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -109,14 +108,26 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
             RequestContext requestContext, 
             bool forceValidation = false)
         {
-
             Uri authorityUri = authorityInfo.CanonicalAuthority;
             string environment = authorityInfo.Host;
 
             if (authorityInfo.IsInstanceDiscoverySupported)
             {
                 var entry = _userMetadataProvider?.GetMetadataOrThrow(environment, requestContext.Logger);
 
+                //Check if instance discovery endpoint is disabled
+                if (entry == null && requestContext.ServiceBundle.Config.InstanceDiscoveryEndpoint == Constants.InstanceDiscoveryEndpointDisabled)
+                {
+                    //Check if regional discovery provider returns an entry. Regional should not be affected by the disabling of instance discovery endpoint.
+                    entry = await _regionDiscoveryProvider.GetMetadataAsync(authorityUri, requestContext).ConfigureAwait(false);
+
+                    if (entry == null)
+                    {
+                        requestContext.Logger.Info("[Instance Discovery] Skipping Instance discovery because it is disabled. ");
```

Comment:
can we also include logging for "Instance discovery is enabled. Instance discovery will be performed"?